### ISSUE TITLE
unstumpable hat issues

### ISSUE THREAD

**Issue by TotallyARussian** [swampservers/contrib#466]
the unstumpable accessory seems to break jigglebones and player movement animation blending, it also lags behind if it's attached to anything but the bone attachments (eyes, ValveBiped.Anim_Attachment_LH, ValveBiped.Anim_Attachment_RH, etc.). the jigglebones spazz out, and player animations, mainly movement animations, kind of snap instead of smoothly blending whenever you strafe or change directions, it's the only accessory to have those issues, pls fix.

**Comment by WinterPhoenix** [comment]
It's not the only accessory with that issue. It's a bug we've been fighting with for a while now.

Basically, there are some scenarios where the engine will re-setup bones every frame, causing jittery jigglebones, or alternatively there are some situations where the accessory isn't properly "predicted" so it's pulling the *server* position of bones/attachments instead of what you see on the client, causing accessories to "lag behind".

AND there's an issue where accessories will just, detach themselves from playermodels spontaneously in some circumstances.

It's a hard problem to fix. I've spent a decent amount of time on it already.

**Comment by TotallyARussian** [comment]
yeah, come to think of it, i do remember it happening with other accessories but very rarely from time to time, but with unstumpable it just happens all the time, is it coded in differently than other accessories?

**Comment by TotallyARussian** [comment]
another thing i might add, is that the same issues happen with any accessory once you enter a vehicle or take a seat

**Comment by WinterPhoenix** [comment]
Not coded particularly differently, no. We know about vehicles/seats also being a factor...
